modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java

```diff
@@ -110,10 +211,12 @@ public void setNpmRepository(String npmRepository) {
     @Override
     public void processOpts() {
         super.processOpts();
-        supportingFiles.add(new SupportingFile("api.mustache", null, "api.ts"));
+        supportingFiles.add(new SupportingFile("models.mustache", modelPackage().replace('.', File.separatorChar), "models.ts"));
+        supportingFiles.add(new SupportingFile("apis.mustache", apiPackage().replace('.', File.separatorChar), "api.ts"));
```

Comment:
isn't there a collision for the destination filename `api.ts`, since both `apis.mustache` and `index.mustache` are generated to `api.ts`?
anyway I would suggest to rename `api.mustache` to `api-single.mustache` and `apis.mustache` to `api.mustache` in order to avoid confusions.

Reply:
No collision there, the apis.mustache goes into a subfolder, but yes, I can clean that up a bit.

Reply:
@macjohnny 

I'm going to go with this:

api-all.mustache = all of the apis re-exported together (output as /apis/apis.ts)
api-single.mustache = single api (output as /apis/{name}Api.ts)
api.mustache = entry point (output as /api.ts to prevent breaking change)

I thin this should avoid any confusion

Reply:
@macjohnny done.  Just awaiting CI again